go-update/src/main/java/org/reactome/release/goupdate/GoTermInstanceModifier.java

```diff
@@ -135,20 +135,33 @@ public void updateGOInstance(Map<String, Map<String, Object>> goTerms, Map<Strin
 				// match the name in the file or if the existing definition does not match
 				// the one in the file, we update with the new name and def'n, and then set
 				// InstanceOf and ComponentOf to NULL, and those get updated later, from whatever's in the GO file.
-				if ((newName!=null && !newName.equals(oldName))
-					|| (newDefinition != null && !newDefinition.equals(oldDefinition)))
+				if ( (newName!=null && !newName.equals(oldName)) || (newDefinition != null && !newDefinition.equals(oldDefinition)))
 				{
-					String nameUpdate = newName.equals(oldName) ? "" : "\n\tNew name:\t\""+newName+"\"\n\told name:\t\""+this.goInstance.getAttributeValue(ReactomeJavaConstants.name)+"\"";
-					String defnUpdate = newDefinition.equals(oldDefinition) ? "" : "\n\tNew def'n:\t\""+newDefinition+"\"\n\told def'n:\t\""+this.goInstance.getAttributeValue(ReactomeJavaConstants.definition)+"\"";
-					nameOrDefinitionChangeStringBuilder.append("\nChange in name/definition for GO:").append(currentGOID).append(nameUpdate).append(defnUpdate);
-					this.goInstance.setAttributeValue(ReactomeJavaConstants.instanceOf, null);
-					this.adaptor.updateInstanceAttribute(this.goInstance, ReactomeJavaConstants.instanceOf);
-					this.goInstance.setAttributeValue(ReactomeJavaConstants.componentOf, null);
-					this.adaptor.updateInstanceAttribute(this.goInstance, ReactomeJavaConstants.componentOf);
-					this.goInstance.setAttributeValue(ReactomeJavaConstants.name, goTerms.get(currentGOID).get(GoUpdateConstants.NAME));
-					this.adaptor.updateInstanceAttribute(this.goInstance, ReactomeJavaConstants.name);
-					this.goInstance.setAttributeValue(ReactomeJavaConstants.definition, newDefinition);
-					this.adaptor.updateInstanceAttribute(this.goInstance, ReactomeJavaConstants.definition);
+					// Changes for name
+					if (newName!=null && !newName.equals(oldName))
+					{
+						String nameUpdate = newName.equals(oldName) ? "" : "\n\tNew name:\t\""+newName+"\"\n\told name:\t\""+this.goInstance.getAttributeValue(ReactomeJavaConstants.name)+"\"";
```

Comment:
Isn't the first condition of this ternary not possible since the preceding if statement filters against newName equaling oldName?

Reply:
Ummm... yeah I'm gonna look at that. ;)